### PR TITLE
JDK-8301480: Replace NULL with nullptr in os/posix

### PR DESCRIPTION
**Pull request by jdksjolen**
Hi, this PR changes all occurrences of NULL to nullptr for the subdirectory os/posix. Unfortunately the script that does the change isn't perfect, and so we
need to comb through these manually to make sure nothing has gone wrong. I also review these changes but things slip past my eyes sometimes.

Here are some typical things to look out for:

1. No changes but copyright header changed (probably because I reverted some changes but forgot the copyright).
2. Macros having their NULL changed to nullptr, these are added to the script when I find them. They should be NULL.
3. nullptr in comments and logs. We try to use lower case "null" in these cases as it reads better. An exception is made when code expressions are in a comment.

An example of this:

```c++
// This function returns null
void* ret_null();
// This function returns true if *x == nullptr
bool is_nullptr(void** x);
```

Note how `nullptr` participates in a code expression here, we really are talking about the specific value `nullptr`.

Thanks!

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8301480](https://bugs.openjdk.org/browse/JDK-8301480): Replace NULL with nullptr in os/posix

### Reviewers
 * [Coleen Phillimore](https://openjdk.org/census#coleenp) (@coleenp - **Reviewer**) ⚠️ Review applies to [a8f990bb](https://git.openjdk.org/jdk/pull/12316/files/a8f990bb14f07e319f71de353ca0c35f5eb25a82)
 * [David Holmes](https://openjdk.org/census#dholmes) (@dholmes-ora - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/12316/head:pull/12316` \
`$ git checkout pull/12316`

Update a local copy of the PR: \
`$ git checkout pull/12316` \
`$ git pull https://git.openjdk.org/jdk pull/12316/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 12316`

View PR using the GUI difftool: \
`$ git pr show -t 12316`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/12316.diff">https://git.openjdk.org/jdk/pull/12316.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/os/posix/perfMemory_posix.cpp [hunk 1]
```diff
@@ -457,19 +457,19 @@ static char* get_user_name(uid_t uid) {
...
       if (result != 0) {
         warning("Could not retrieve passwd entry: %s\n",
                 os::strerror(result));
       }
-      else if (p == NULL) {
+      else if (p == nullptr) {
         // this check is added to protect against an observed problem
         // with getpwuid_r() on RedHat 9 where getpwuid_r returns 0,
-        // indicating success, but has p == NULL. This was observed when
+        // indicating success, but has p == null. This was observed when
```
Comment (jdksjolen) [inline src/hotspot/os/posix/perfMemory_posix.cpp:472]: nullptr

**Review by jdksjolen — COMMENTED** [review]
Found 1issue

**Review by coleenp — APPROVED** [review]
trivially fine.

File: src/hotspot/os/posix/signals_posix.cpp [hunk 2]
```diff
@@ -1337,14 +1337,14 @@ void install_signal_handlers() {
...
-  // EXC_MASK_BAD_ACCESS needed by all architectures for NULL ptr checking
+  // EXC_MASK_BAD_ACCESS needed by all architectures for null ptr checking
   // EXC_MASK_ARITHMETIC needed by all architectures for div by 0 checking
   // EXC_MASK_BAD_INSTRUCTION needed by aarch64 to initiate deoptimization
   kern_return_t kr;
   kr = task_set_exception_ports(mach_task_self(),
                                 EXC_MASK_BAD_ACCESS | EXC_MASK_ARITHMETIC
                                   AARCH64_ONLY(| EXC_MASK_BAD_INSTRUCTION),
-                                MACH_PORT_NULL,
+                                MACH_PORT_nullptr,
```
Comment (dholmes-ora) [inline src/hotspot/os/posix/signals_posix.cpp:1347]: This is wrong. Needs reverting. (How did this build on macOS?)

**Review by dholmes-ora — CHANGES_REQUESTED** [review]
Generally good but one bug. Thanks

**Review by dholmes-ora — APPROVED** [review]
Update looks good. Thanks.

**Comment by jdksjolen** [comment]
Passes tier1, integrating.

/integrate

**Pull request closed without merging**